src/main/java/racingcargame/model/Car.java

```diff
@@ -0,0 +1,67 @@
+package racingcargame.model;
+
+import java.util.Objects;
+
+public class Car implements Comparable<Car> {
+    private static final int MAXIMUM_NAME_LENGTH = 5;
+
+    private final String name;
+    private int position = 0;
+
+    public Car(String name) {
+        checkNameLength(name);
+        this.name = name;
+    }
+
+    public Car(String name, int position) {
+        this.name = name;
+        this.position = position;
+    }
+
+    private static void checkNameLength(String name) {
+        if (name.isEmpty() || name.length() > MAXIMUM_NAME_LENGTH) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getPosition() {
+        return position;
+    }
+
+    public Car update(int randomNumber) {
+        if (randomNumber > 4) {
+            return new Car(name, position + 1);
+        }
+        return new Car(name, position);
```

Comment:
`Car`가 VALUE OBJECT라면 아래의 코드가 가능합니다. 왜 그럴까요?

```suggestion
        return this;
```

Reply:
혹시 Value Object라면 값이 바뀔때 새로운 객체를 생성해주어서 그런 것인가요?